### PR TITLE
Fb/edit track

### PR DESCRIPTION
**Pull request by rlindner81**

### CODE REVIEW

File: frontend/src/components/AddOrEditStep.vue [hunk 1]
```diff
@@ -1,75 +1,104 @@
...
+const emit = defineEmits(["close"]);
+
+// NOTE: this is a little weird, but props.show is _not_ reactive, so we cannot watch it
+//   directly and wrapping it in toRef would be misleading, since its readonly, so
+//   computed is a fair compromise
+const show = computed(() => props.show);
+
+watch(show, (newValue) => {
+  if (newValue) {
+    if (!props.edit) {
```
Comment (Phantomkeks) [inline frontend/src/components/AddOrEditStep.vue:31]: Would it make sense to merge this to `if`s?

File: frontend/src/components/AddOrEditStep.vue [hunk 2]
```diff
@@ -1,75 +1,104 @@
...
-    v-model="showAddStepModal"
-    persistent
-    v-if="trackStore.current && stepStore.newStepValues && stepStore.newStepEnabled"
-  >
+  <v-dialog v-model="show" persistent>
     <v-card class="pa-2">
-      <v-card-title class="text-h5">{{ $t("entity.step.add") }}</v-card-title>
-      <v-container>
+      <v-card-title class="text-h5">{{ edit ? $t("entity.step.edit") : $t("entity.step.add") }}</v-card-title>
+      <!-- TODO v-else would be nice-->
```
Comment (Phantomkeks) [inline frontend/src/components/AddOrEditStep.vue:55]: Does it make sense to have this as a GitHub issue or are inline `todo`s the more preferred way?

Comment (rlindner81) [reply frontend/src/components/AddOrEditStep.vue:55]: For this one, we can keep the inline TODO. The project is not in a state where I would go over it with this fine granularity. Going back over my code, I felt like the data modelling has some problems, but more than that the whole vue ui coding approach that I adopted here feels wrong to me. 
I want to take some Mondays and explore other frameworks, like Svelte or Leptos, and see if these feels better for this type of project.

File: frontend/src/components/AddOrEditStep.vue (hunk 1 above)
Comment (rlindner81) [reply frontend/src/components/AddOrEditStep.vue:31]: Yeah totally, this is just a leftover. The inner if used to be in it's own function. I will move them together.

**Pull request merged**